File: 模拟/贪心/将字符串翻转到单调递增/将字符串翻转到单调递增.cpp
// https://leetcode.cn/problems/flip-string-to-monotone-increasing/

class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int num0 = 0, num1 = 0, ans = 1e6;
        for (char c : s) if (c == '0') num0 ++;
        for (char c : s)
        {
            ans = min(ans, num0 + num1);
            if (c == '1') num1 ++;
            else num0 --;
        }
        ans = min(ans, num0 + num1);
        return ans;
    }
};

File: 模拟/贪心/将字符串翻转到单调递增/将字符串翻转到单调递增.py
# https://leetcode.cn/problems/flip-string-to-monotone-increasing/

class Solution:
    def minFlipsMonoIncr(self, s: str) -> int:
        num0, num1, ans = s.count('0'), 0, 1000000
        ans = min(ans, num0 + num1)
        for i in s:
            if i == '0': num0 -= 1
            else: num1 += 1
            ans = min(ans, num0 + num1)
        return ans
        